### PR TITLE
parse union inside cte

### PR DESCRIPTION
**Pull request by duongphannamhung**
fix issues #376 
commit [bd16d45](https://github.com/reata/sqllineage/pull/391/commits/bd16d45b885511e7b3d3f6b4585f333d3d69fa6e): add check union direct inside cte, we current just have check union in raw segment
commit [94672b3](https://github.com/reata/sqllineage/pull/391/commits/94672b33bbd93aaf253d195f479be322a6b1852e): use union direct method to check union inside cte, then add it to queue for parsing
commit [d8eb39d](https://github.com/reata/sqllineage/pull/391/commits/d8eb39df4f3d094df7ee6dd29f9d1f81cf7aab02): add test for this issue

### CODE REVIEW

**Comment by reata** [comment]
Would you fix the CI? You can run it in your local env via `tox -e py38`. Or change to your local Python version like py39 or py310. You can rely on GitHub Actions for multiple Python version test though if you don't have to set it up locally.

**Comment by duongphannamhung** [comment]
I just fix the CI, thank you

File: tests/test_others_dialect_specific.py
Comment (reata) [inline tests/test_others_dialect_specific.py:1]: This is duplicated test case. Let's remove it and keep it only in test_columns.py

File: sqllineage/core/parser/sqlfluff/extractors/cte_extractor.py [hunk 1]
```diff
@@ -70,8 +74,15 @@ def extract(
                         identifier = sub_segment.raw
                         if not segment_has_alias:
                             holder.add_cte(SqlFluffSubQuery.of(sub_segment, identifier))
+
+                    parse_sub = self.parse_subquery(sub_segment)
+                    if is_union_direct(sub_segment):
+                        for sm in sub_segment.segments:
+                            if sm.type == "set_expression":
+                                parse_sub.append(SqlFluffSubQuery.of(sm, None))
+
```
Comment (reata) [inline sqllineage/core/parser/sqlfluff/extractors/cte_extractor.py:83]: I think we can move this part under `if sub_segment.type == "bracketed":`

File: sqllineage/core/parser/sqlfluff/utils.py [hunk 2]
```diff
@@ -413,6 +413,13 @@ def is_union(segment: BaseSegment) -> bool:
     )
 
 
+def is_union_direct(segment: BaseSegment) -> bool:
+    """
+    like is_union but use segments rather than raw_segments
+    """
+    return len([s for s in segment.segments if s.type == "set_expression"]) > 0
+
+
```
Comment (reata) [inline sqllineage/core/parser/sqlfluff/utils.py:422]: Any particular reason we had to add this. The test case would also pass using `is_union`

**Review by reata — CHANGES_REQUESTED** [review]
Please address my comment inline. Also let me know if you're interested in making it work with sqlparse as well.

**Comment by duongphannamhung** [comment]
Really appreciate your comments, to be `is_isomorphic` like sqlparse makes me rethink the way my commit work so I have a little few changes, hope it's working out

**Pull request merged**